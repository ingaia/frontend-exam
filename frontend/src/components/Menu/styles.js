import styled from 'styled-components';
import { lighten } from 'polished';

export const MenuTop = styled.div`
    display: flex;
    flex-direction: row;
    border-bottom: 1px solid #4c4c4c;
    justify-content: space-between;
    align-items: center;
    padding-top: 20px;
    padding-bottom: 10px;
    background-color: #000;
    img {
        margin-left: 20px;
        width: 124px;
        height: 77px;
    }

    @media (min-width: 992px) {
        display: none;
    }
`;

export const MenuList = styled.div`
    display: flex;
    flex-direction: row;
    align-items: center;
    margin-right: 20px;
    @media (max-width: 400px) {
        display: flex;
        flex-direction: column;
    }

    button {
        width: 154px;
        height: 39px;
        background: transparent;
        border: 1px solid #a99e7e;
        font-size: 14px;
        line-height: 16px;
        color: #a99e7e;
        transition: background 0.2s;
        margin: 10px 10px;

        &:hover {
            background: ${lighten(0.9, 'rgba(255, 255, 255, 0.1)')};
        }
    }

    span {
        font-size: 14px;
        line-height: 16px;
        color: #fff;
        &:hover {
            cursor: pointer;
        }
    }
`;
