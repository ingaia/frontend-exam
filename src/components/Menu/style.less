@import './../../variables.less';

.menu {
    flex: 2;
    height: 100%;
    
    .box-menu {
        display: flex;
        flex-direction: column;
        align-items: center;
        padding: 10px;
        margin-top: 150px;
    }

    .logo {
        width: 275px;
        margin-bottom: 100px;
    }

    .btn {
        padding: 10px 100px;
    }
}

@media @mobile {
    .menu {
        display: none;
    }
}