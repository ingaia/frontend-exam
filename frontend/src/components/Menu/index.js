import React from 'react';

import { MenuTop, MenuList } from './styles';

import logo from '~/assets/logo_dark.png';

export default function Menu() {
    return (
        <MenuTop>
            <img src={logo} alt="logo" />
            <MenuList>
                <button type="button">TRAILERS</button>
                <span>LOGOUT</span>
            </MenuList>
        </MenuTop>
    );
}
