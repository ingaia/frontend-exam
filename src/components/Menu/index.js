import React from 'react';
import Button from '../../components/Button';

import logo_dark from '../../../assets/logo_dark.png';
import './style.less';

export default function Menu(props){
    return (
        <div className="menu">
            <div className="box-menu">
                <img className="logo" src={logo_dark} alt=""/>
                <Button className="btn-hillary" label="Trailers"/>
                <Button className="btn-default" label="Logout" onClick={props.logOut}/> 
            </div>
        </div>
    );
}