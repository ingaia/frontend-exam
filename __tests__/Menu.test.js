import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'
import Menu from '../src/components/Menu';

describe('<Menu/>', () => {
    it('should render', () => {
        const wrapper = shallow(<Menu/>);

        expect(wrapper.hasClass('menu')).toBe(true);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should call the function logOut', () => {
        const logOut = jest.fn();
        const wrapper = shallow(<Menu logOut={logOut}/>);

        wrapper.find('.btn-default').simulate('click');

        expect(logOut).toHaveBeenCalled();
    });
});