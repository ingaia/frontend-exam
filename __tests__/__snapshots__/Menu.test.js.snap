// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu/> should render 1`] = `
<div
  className="menu"
>
  <div
    className="box-menu"
  >
    <img
      alt=""
      className="logo"
      src="test-file-stub"
    />
    <Button
      className="btn-hillary"
      label="Trailers"
    />
    <Button
      className="btn-default"
      label="Logout"
    />
  </div>
</div>
`;
